app/javascript/components/key-performance-indicators/components/as-key-performance-indicator/component.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
+import { Help } from "@material-ui/icons";
+import makeStyles from "@material-ui/styles/makeStyles";
+
+import OptionsBox from "../../../dashboard/options-box";
+import { useI18n } from "../../../i18n";
+import actionsForKPI from "../../action-creators";
+import selectorsForKPI from "../../selectors";
+import DateRangeSelect from "../date-range-select";
+import Permission from "../../../application/permission";
+import { RESOURCES } from "../../../../libs/permissions";
+import usePermissions from "../../../permissions";
+
+import styles from "./styles.css";
+
+const asKeyPerformanceIndicator = (identifier, defaultData, action) => {
+  return Visualizer => {
+    const enhance = connect(state => ({ data: selectorsForKPI(identifier, state, defaultData) }), {
+      fetchData: actionsForKPI(identifier)
+    });
+
+    const Component = enhance(({ data, fetchData, dateRanges, ...props }) => {
+      const i18n = useI18n();
+      const css = makeStyles(styles)();
+      const canViewKpi = usePermissions(RESOURCES.kpis, [action]);
+
+      const [currentDateRange, setCurrentDateRange] = useState(dateRanges[0]);
+
+      useEffect(() => {
+        if (canViewKpi) fetchData(currentDateRange);
+      }, [currentDateRange, canViewKpi]);
+
+      const [helptextOpen, setHelptextOpen] = useState(false);
+      const handleHelptextClick = () => setHelptextOpen(!helptextOpen);
+
+      return (
+        <Permission resources={RESOURCES.kpis} actions={[action]}>
+          <OptionsBox
+            title={i18n.t(`key_performance_indicators.${identifier}.title`)}
+            action={
+              <DateRangeSelect
+                i18n={i18n}
+                ranges={dateRanges}
+                selectedRange={currentDateRange}
+                setSelectedRange={setCurrentDateRange}
+                withCustomRange
+              />
+            }
+          >
+            <Visualizer identifier={identifier} data={data} {...props} />
+            <div>
+              <div title={i18n.t(`key_performance_indicators.helptext.helptext`)} className={css.helptextHeader}>
+                <Help className={css.helptextButton} onClick={handleHelptextClick} />
+              </div>
+              {helptextOpen && (
+                <p className={css.helptextBody}>{i18n.t(`key_performance_indicators.${identifier}.helptext`)}</p>
+              )}
+            </div>
+          </OptionsBox>
+        </Permission>
+      );
+    });
+
+    Component.identifier = identifier;
```

Comment:
I don't get these 2 lines. Is this who you are passing props?

Reply:
I was using those for some fancy permissions stuff, conditionally showing parts of the KPI page but I didn't want them to have to hard code which KPIs would be present so I was adding some info to the generated KPI components which could then be queried later by these sections.

It got a bit complicated so I removed the logic and this survived. I think I can just remove this.

Reply:
OK, I just couldn't figure out how it was used